macros/commands/generate_json_schema_test.sql

```diff
@@ -0,0 +1,101 @@
+      {% do return(none) %}
+    {% endif %}
+
+    {% set elementary_database_name, elementary_schema_name = elementary.get_package_database_and_schema() %}
+
+    {% do node.config.update({"packages": ["genson"]}) %}
+    {% do node.update({'database': elementary_database_name, 'schema': elementary_schema_name}) %}
+    {% if node.resource_type == 'source' %}
+      {# Source nodes don't have alias, and submit_python_job expects it #}
+      {% do node.update({'alias': "jsonschemagen_{}_{}".format(node.source_name, node.name)}) %}
+    {% endif %}
+
+    {% set output_table = api.Relation.create(database=elementary_database_name, schema=elementary_schema_name,
+        identifier='json_schema_tmp__' ~ node.alias).quote(false, false, false) %}
+
+    {% set gen_json_schema_func = elementary.generate_json_schema_py_func(column_name) %}
+    {% set node_relation = node_relation.quote(false, false, false) %}
+    {% set compiled_py_code = adapter.dispatch('compile_py_code', 'elementary')(node_relation, gen_json_schema_func,
+                                                                                output_table, code_type='function') %}
+
+    {% do elementary.run_python(node, compiled_py_code) %}
+    {% set json_schema = elementary.result_value('select result from {}'.format(output_table)) %}
+    {% if json_schema == 'genson_not_installed' %}
+      {% do print("ERROR - the 'genson' python library is missing from your warehouse.") %}
+      {% do print("") %}
+      {% do print("This macro relies on the 'genson' python library for generating JSON schemas. Please follow dbt's instructions here: ") %}
+      {% do print("https://docs.getdbt.com/docs/building-a-dbt-project/building-models/python-models#specific-data-warehouses") %}
+      {% do print("regarding how to install python packages for a {} warehouse.".format(target.type)) %}
+      {% do return(none) %}
+    {% endif %}
+
+    {% if not json_schema %}
+        {% do print("Not a valid JSON column: {}".format(column_name)) %}
+        {% do return(none) %}
+    {% endif %}
+
+    {% set json_schema = fromjson(json_schema) %}
+    {% do json_schema.pop('$schema', None) %}
+
+    {% set testyaml %}
+columns:
+  - name: {{ column_name }}
+    tests:
+      - elementary.json_schema:
+          {{ toyaml(json_schema) | indent(10) }}
+    {% endset %}
+
+    {% do print("Please add the following test to your {} configuration for the column {}:".format(node.resource_type, column_name)) %}
+    {% do print(testyaml) %}
+{% endmacro %}
+
+{% macro generate_json_schema_py_func(column_name) %}
+import json
+try:
+    import genson
+except ImportError:
+    genson = None
+
+def get_column_name_in_df(df, column):
+    matching = [col for col in df.columns if col.lower() == column.lower()]
+    if len(matching) == 0:
+        raise Exception("No column with the name: {}".format(col))
+    elif len(matching) > 1:
+        # If there's more than one match, then case matters, so return as-is
```

Comment:
Not sure if it matters, but technically this still doesn't mean that there was a match.
I can search for `heLLo` and `Hello` and `hellO` matched, and I'd return `heLLo` regardless.

Reply:
Hmm yeah you're right
Maybe I should just fail here

Reply:
Or I can do a case-sensitive search in that case

Reply:
But it's an edge case hh

Reply:
Yeah, generally I wouldn't have pointed this out, just sharing since we already consider the casing.
I think we can just look for a lower-case match and that should be enough for now.

Reply:
Fixed to simpler logic that is more correct (I think)